core/includes/form.inc

```diff
@@ -3629,17 +3629,23 @@ function form_process_container($element, &$form_state) {
  */
 function theme_container($variables) {
   $element = $variables['element'];
-  // Ensure #attributes is set.
-  $element += array('#attributes' => array());
+  // Ensure #attributes is set, but do not overwrite them if they have already
+  // been set.
+  $attributes = isset($element['#attributes']) ? $element['#attributes'] : array();
+  $element += array('#attributes' => $attributes);
 
   // Special handling for form elements.
   if (isset($element['#array_parents'])) {
-    // Assign an html ID.
+    // Assign an HTML ID.
     if (!array_key_exists('id', $element['#attributes'])) {
       $element['#attributes']['id'] = $element['#id'];
     }
-    // Add the form-wrapper class.
-    if (empty($element['#attributes']['class'])) {
+    // Ensure that the classes within #attributes have been initialized.
+    if (!array_key_exists('class', $element['#attributes'])) {
+      $element['#attributes']['class'] = array();
+    }
+    // Add the form-wrapper class if not already added.
+    if (!in_array('form-wrapper', $element['#attributes']['class'])) {
```

Comment:
Not sure why we were previously adding this class only when there were no other classes. Doesn't make sense to me at all 🤷🏼

Reply:
...this is why: https://github.com/backdrop/backdrop-issues/issues/711

We need to test and make sure that that issue has not regressed with this change here